src/edu/csus/ecs/pc2/ui/TeamStatusPane.java

```diff
@@ -321,25 +326,65 @@ private void repopulateGrid(boolean populate) {
             teamLabel.setToolTipText(toolTipText);
             centerPane.add(teamName, teamLabel);
         }
+        
+        
+        /**
+         * Update counts for each status
+         */
+        
+        submittedClarsLabel.setText("Submitted Clar(s) (" + toInt( teamStatusCount.get(TeamStatus.TEAM_HAS_SUBMITTED_CLARS_ONLY), 0) + ")");
+        submittedRunsLabel.setText("Submitted Run(s) (" +toInt(  teamStatusCount.get(TeamStatus.TEAM_HAS_SUBMITTED_RUNS_ONLY) , 0) + ")");
+        hasLoggedInLabel.setText("Has Logged In (" + toInt( teamStatusCount.get(TeamStatus.TEAM_HAS_LOGGED_IN), 0 ) + ")");
+        nocontactLabel.setText("No Contact (" + toInt( teamStatusCount.get(TeamStatus.NO_TEAM_CONTACT), 0) + ")");
+        readyLabel.setText("READY (" + toInt( teamStatusCount.get(TeamStatus.TEAM_HAS_SUBMITTED_RUNS_AND_CLARS), -0) + ")");
+
         centerPane.repaint();
     }
 
+    private int toInt(Integer integer, int defaultValue) {
+        if (integer == null) {
+            return defaultValue;
+        } else {
+            return integer.intValue();
+        }
+    }
+
     private Color getStatusColor(ClientId clientId, Run[] runs, Clarification[] clarifications) {
         Color outColor = NO_CONTACT_COLOR;
 
         if (runs.length > 0 && clarifications.length > 0) {
             outColor = HAS_SUBMITTED_RUNS_AND_CLARS_COLOR;
+            incrementStatusCount(TeamStatus.TEAM_HAS_SUBMITTED_RUNS_AND_CLARS);
```

Comment:
It seems a bit awkward to have a method whose title is `getStatusCOLOR` having a side-effect like incrementing a status COUNT...

Reply:
Again, do you have a suggestion or just a remark ?

Reply:
Two suggested changes:  (1) remove code from method `getStatusColor()` which CHANGES state (getters should only return state, not change it); and (2) create a new method named like `updateStatusCount(TeamStatus )` which increments the specified team status value.

Reply:
pushed fix.